pkg/controller/rds/dbinstance/setup.go

```diff
@@ -0,0 +1,190 @@
+	name := managed.ControllerName(svcapitypes.DBInstanceGroupKind)
+	opts := []option{
+		func(e *external) {
+			c := &custom{client: e.client, kube: e.kube}
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.preCreate = c.preCreate
+			e.postCreate = c.postCreate
+			e.preDelete = preDelete
+			e.filterList = filterList
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.DBInstance{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.DBInstanceGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.RDSAPI
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.DescribeDBInstancesInput) error {
+	obj.DBInstanceIdentifier = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func (e *custom) preCreate(ctx context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.CreateDBInstanceInput) error {
+	pw, _, err := rds.GetPassword(ctx, e.kube, cr.Spec.ForProvider.MasterUserPasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if resource.IgnoreNotFound(err) != nil {
+		return errors.Wrap(err, "cannot get password from the given secret")
+	}
+	if pw == "" && cr.Spec.ForProvider.AutogeneratePassword {
+		pw, err = password.Generate()
+		if err != nil {
+			return errors.Wrap(err, "unable to generate a password")
+		}
+		if err := e.savePasswordSecret(ctx, cr, pw); err != nil {
+			return errors.Wrap(err, errSaveSecretFailed)
+		}
+	}
+	obj.MasterUserPassword = aws.String(pw)
+	obj.DBInstanceIdentifier = aws.String(meta.GetExternalName(cr))
+	obj.VpcSecurityGroupIds = make([]*string, len(cr.Spec.ForProvider.VPCSecurityGroupIDs))
+	for i, v := range cr.Spec.ForProvider.VPCSecurityGroupIDs {
+		obj.VpcSecurityGroupIds[i] = aws.String(v)
+	}
+	obj.DBSecurityGroups = make([]*string, len(cr.Spec.ForProvider.DBSecurityGroups))
+	for i, v := range cr.Spec.ForProvider.DBSecurityGroups {
+		obj.DBSecurityGroups[i] = aws.String(v)
+	}
+	return nil
+}
+
+func (e *custom) postCreate(ctx context.Context, cr *svcapitypes.DBInstance, _ *svcsdk.CreateDBInstanceOutput, _ managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	conn := managed.ConnectionDetails{
+		xpv1.ResourceCredentialsSecretUserKey: []byte(aws.StringValue(cr.Spec.ForProvider.MasterUsername)),
+	}
+	pw, _, err := rds.GetPassword(ctx, e.kube, cr.Spec.ForProvider.MasterUserPasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if resource.IgnoreNotFound(err) != nil {
```

Comment:
```suggestion
	if err != nil {
```

We shouldn't actually ignore this error since that's the only way user can provide a password.